File: implementation/apple-and-orange/apple-and-orange.go
package main

import "fmt"

func between(x, left, right int) bool {
	return x >= left && x <= right
}

func main() {
	var houseLeft, houseRight, appleTree, orangeTree, appleCount, orangeCount, distance, applesHit, orangesHit int
	var appleHitMin, appleHitMax, orangeHitMin, orangeHitMax int
	fmt.Scanln(&houseLeft, &houseRight)
	fmt.Scanln(&appleTree, &orangeTree)
	fmt.Scanln(&appleCount, &orangeCount)
	appleHitMin = houseLeft - appleTree
	appleHitMax = houseRight - appleTree
	orangeHitMin = houseLeft - orangeTree
	orangeHitMax = houseRight - orangeTree
	for i := 0; i < appleCount; i++ {
		fmt.Scan(&distance)
		if between(distance, appleHitMin, appleHitMax) {
			applesHit++
		}
	}
	for i := 0; i < orangeCount; i++ {
		fmt.Scan(&distance)
		if between(distance, orangeHitMin, orangeHitMax) {
			orangesHit++
		}
	}
	fmt.Println(applesHit)
	fmt.Println(orangesHit)
}

File: implementation/apple-and-orange/apple-and-orange.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {
    
    static boolean between(int x, int left, int right) {
        return x >= left && x <= right;
    }

    public static void main(String[] args) {
        int applesHit = 0;
        int orangesHit = 0;
        
        Scanner in = new Scanner(System.in);
        int s = in.nextInt();
        int t = in.nextInt();
        int a = in.nextInt();
        int b = in.nextInt();
        int m = in.nextInt();
        int n = in.nextInt();
        int appleMin = s - a;
        int appleMax = t - a;
        int orangeMin = s - b;
        int orangeMax = t - b;

        int distance;
        for(int i=0; i < m; i++){
            distance = in.nextInt();
            if (between(distance, appleMin, appleMax)) {
                applesHit += 1;
            }
        }

        for(int i=0; i < n; i++){
            distance = in.nextInt();
            if (between(distance, orangeMin, orangeMax)) {
                orangesHit += 1;
            }
        }
        
        System.out.println(applesHit);
        System.out.println(orangesHit);
    }
}